src/Identifier.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Equip\ValueObject;
+
+use InvalidArgumentException;
+
+class Identifier
```

Comment:
I would rather non-integers be called `StringIdentifier` or `UuidIdentifier`, etc.